changelog.md

```diff
@@ -108,18 +99,221 @@ deliver information you explicitly requested by default, even though we will
+
+As a result of this, the `--max_tasks_per_offer_cycle` options is deprecated and has no effect anymore.
+
+To prevent overloading Marathon and maintain speedy offer processing, there is also a timeout for matching each
+incoming resource offer, i.e. finding suitable tasks to launch for incoming offers.
+
+* `--offer_matching_timeout` (Optional. Default: 1000): 
+    Offer matching timeout (ms). Stop trying to match additional tasks for this offer after this time.
+    All already matched tasks are launched.
+
+
+#### No pseudo-deterministic assignment of host ports anymore
+
+If you specify non-zero `"ports"` in your app JSON, they are used as service ports. The old code contained logic that 
+would assign these ports as host ports if available in the
+processed offer. That misled people into thinking that these ports corresponded to host ports. The new code always
+randomizes host ports assignment without `"requirePorts"` or explicit `"hostPort"` configuration.
+
+#### Last task failures are persisted
+
+Marathon has exposed the `"lastTaskFailure"` via the API for a while but this information was not persisted
+ across restarts or on fail over. Now it is.
+ 
+#### Logging command line parameters on startup
+
+Marathon will now log all command line parameters on startup. Example:
+
+```
+2015-08-02 11:48:08,047] INFO Starting Marathon 0.11.0-SNAPSHOT with --zk zk://master.mesos:2181/marathon --master master.mesos:5050
+```
+
+#### Improved logging of deployment plans
+
+Example:
+
+```
+DeploymentPlan 2015-09-02T07:56:12.105Z
+step 1:
+  * Start(App(app1, cmd="cmd")), instances=0)
+step 2:
+  * Scale(App(app1, cmd="cmd")), instances=2)
+```
+
+#### Improved logging of offer rejections
+
+Example for missing port:
+
+```
+[2015-08-04 20:11:20,510] INFO Offer [20150804-173037-16777343-5050-4340-O110]. Cannot find range with host port 8080 for app [/product/frontend]
+[2015-08-04 20:11:29,422] INFO Offer [20150804-173037-16777343-5050-4340-O110]. Insufficient resources for [/product/frontend] (need cpus=1.0, mem=64.0, disk=1.0, ports=([8080] required + 1 dynamic), available in offer:
+...
+```
+
+Example for missing basic resources:
+
+```
+[2015-08-04 20:17:27,986] INFO Offer [20150804-173037-16777343-5050-4340-O110]. Not all basic resources satisfied: cpu NOT SATISFIED (30.0 > 8.0), disk SATISFIED (0.0 <= 0.0), mem SATISFIED (16.0 <= 15360.0)
+[2015-08-04 20:17:27,989] INFO Offer [20150804-173037-16777343-5050-4340-O110]. Insufficient for [/test] (need cpus=30.0, mem=16.0, disk=0.0, ports=(1 dynamic), available in offer:
+...
+```
+
+#### Immediately store changed apps
+
+In the past, Marathon would update the data for the group endpoint when it accepted a new deployment request
+like a change an app configuration. Marathon would only update the data in the app endpoint when it started
+the related deployment step. All in all, that was confusing for our users and thus we will now update both
```

Comment:
`All in all` is a slightly odd switch of tone. How about
`We found that this was confusing...`